gadget/gadget.go

```diff
@@ -69,6 +71,14 @@ type Info struct {
 	Connections []Connection `yaml:"connections"`
 }
 
+// ModelConstraints defines rules to be followed when reading the gadget metadata.
+type ModelConstraints struct {
+	// Classic rules (i.e. content/presence of gadget.yaml is fully optional)
+	Classic bool
+	// System seeding is enabled (Core 20)
+	SystemSeed bool
```

Comment:
NeedsSystemSeed?

Reply:
I think it's more in the sense of "has" than "needs". When we have the `system-seed` role the new core20 rules apply, otherwise stick to the old core/core18 rules.